internal/provider/bootstrap.go

```diff
@@ -76,8 +76,18 @@ func Bootstrap(e *pluggable.Event) pluggable.EventResponse {
 		return pluggable.EventResponse{}
 	}
 
+	if (tokenNotDefined && prvConfig.K0s.Enabled) || prvConfig.K0sWorker.Enabled || skipAuto {
+		err := oneTimeBootstrap(logger, prvConfig, func() error {
+			return SetupVPN(services.EdgeVPNDefaultInstance, cfg.APIAddress, "/", true, prvConfig)
+		})
+		if err != nil {
+			return ErrorEvent("Failed setup: %s", err.Error())
+		}
+		return pluggable.EventResponse{}
+	}
```

Comment:
this whole section is about vpn setup, I think instead of having a block that mostly repeats itself, we could have it all in the previous `if` something along the lines of:

```
if (tokenNotDefined && (prvConfig.K3s.Enabled || prvConfig.K3sAgent.Enabled) || (prvConfig.K0s.Enabled || prvConfig.K0sWorker.Enabled) || skipAuto {
```

Reply:
@kairos-io/maintainers I think we should implement more functions so code like this would read a lot more easily something like 

```
if tokenNotDefined && (k3sEnabled() || k0sEnabled()) ...
```

But for some reason we like the verbosity of golang :P ... I think a bit "cleaner" code (to use Uncle Bob's naming, not saying this is not clean) would benefit contributions

Reply:
this is just an internal comment doesn't need to be addressed before merging